applications/poisson/sine/application.h

```diff
@@ -216,9 +217,44 @@ class Application : public ApplicationBase<dim, n_components, Number>
       }
     };
 
-    this->grid->create_triangulation(this->param.grid,
-                                     lambda_create_coarse_triangulation,
-                                     this->param.grid.n_refine_global);
+    auto const lambda_read_external_coarse_grid = [&](dealii::Triangulation<dim, dim> & tria) {
+      AssertThrow(!this->param.grid.grid_file.empty(),
+                  dealii::ExcMessage(
+                    "You are trying to read a grid file, but the the string is empty."));
+
+      dealii::GridIn<dim> grid_in;
+
+      grid_in.attach_triangulation(tria);
+
+      std::string extension =
+        this->param.grid.grid_file.substr(this->param.grid.grid_file.find_last_of('.') + 1);
+
+      AssertThrow(!extension.empty(), dealii::ExcMessage("File extension was empty!"));
```

Comment:
"was" -> "is" would be better in my opinion. Again, the message is a bit unspecific. Since this functionality will become some utility function in ExaDG, the description should be more detailed.

Reply:
changed in d932484523cc4b0a6ccbe41963e79a51e5d74191.